src/contentScript/contentScriptPlatform.test.ts

```
@@ -0,0 +1,135 @@
+    appApiMock.onGet("/api/me/").reply(200, { flags: [] });
+
+    const config = sanitizedIntegrationConfigFactory();
+    const request = { url: "https://example.com" };
+    await contentScriptPlatform.request(config, request);
+
+    expect(backgroundRequestMock).toHaveBeenCalledOnce();
+    // `toHaveBeenCalledExactlyOnceWith`'s matching is confusing
+    expect(backgroundRequestMock).toHaveBeenCalledWith(config, request, {
+      interactiveLogin: true,
+    });
+  });
+
+  it("makes non-interactive successful call", async () => {
+    appApiMock
+      .onGet("/api/me/")
+      .reply(200, { flags: ["integration-login-banner"] });
+
+    const config = sanitizedIntegrationConfigFactory();
+    const request = { url: "https://example.com" };
+    await contentScriptPlatform.request(config, request);
+
+    expect(backgroundRequestMock).toHaveBeenCalledOnce();
+    // `toHaveBeenCalledExactlyOnceWith`'s matching is confusing
+    expect(backgroundRequestMock).toHaveBeenCalledWith(config, request, {
+      interactiveLogin: false,
+    });
+  });
+
+  it("handles interactive error", async () => {
+    appApiMock
+      .onGet("/api/me/")
+      .reply(200, { flags: ["integration-login-banner"] });
+
+    backgroundRequestMock.mockRejectedValueOnce(new Error("Other Error"));
+
+    const config = sanitizedIntegrationConfigFactory();
+    const request = { url: "https://example.com" };
+    await expect(
+      contentScriptPlatform.request(config, request),
+    ).rejects.toThrow();
+
+    expect(backgroundRequestMock).toHaveBeenCalledOnce();
+    // `toHaveBeenCalledExactlyOnceWith`'s matching is confusing
+    expect(backgroundRequestMock).toHaveBeenCalledWith(config, request, {
+      interactiveLogin: false,
+    });
+  });
+
+  it("handles deferred login", async () => {
+    appApiMock
+      .onGet("/api/me/")
+      .reply(200, { flags: ["integration-login-banner"] });
+
+    backgroundRequestMock.mockRejectedValueOnce(
```

We're testing here that the login banner is displayed if the request receives an InteractiveLoginRequiredError, but I don't believe we have any tests anywhere that validate that when `launchWebFlow` throws that error, it isn't swallowed between there and `performConfiguredRequest`.

---

Added a test in performConfiguredRequest.test.ts